test.py

```diff
@@ -563,16 +563,65 @@ def run_test(framework, cluster, repo_root):
         logger.info(msg, runtests_script, framework.name)
         raise CommandFailure(cmd_args)
 
+def setup_junit_xml():
+    """Try to download and import junit_xml"""
+    workdir = get_work_dir()
+    lib_dir = os.path.join(workdir, 'lib', 'python')
+    pip_cmd = ['pip', 'install', 'junit-xml', '-t', lib_dir]
```

Comment:
It's slightly confusing that this script modifies its own execution environment in a subprocess and then imports on the basis of that change. Could we group all this environment mutation into one script and the actual test logic (`emit_junit_xml`) in another?